client/my-sites/plans/jetpack-plans/product-store/index.tsx

```diff
@@ -21,7 +21,12 @@ const ProductStore: React.FC< ProductStoreProps > = ( {
 	const siteId = useSelector( getSelectedSiteId );
 	const productSlugs = useProductSlugs( { siteId, duration } );
 
-	const [ currentView, setCurrentView ] = useState< ViewType >( 'products' );
+	const defaultViewType =
+		urlQueryArgs?.view && [ 'products', 'bundles' ].includes( urlQueryArgs.view )
+			? urlQueryArgs.view
+			: 'products';
+
+	const [ currentView, setCurrentView ] = useState< ViewType >( defaultViewType );
```

Comment:
What about moving this computation inside a callback to `useState`?

```suggestion
	const [ currentView, setCurrentView ] = useState< ViewType >( () => {
		return urlQueryArgs?.view && [ 'products', 'bundles' ].includes( urlQueryArgs.view )
			? urlQueryArgs.view
			: 'products';
	} );
```

Reply:
The results of this computation is required in other components after the recent commit, so not moving it into callback